keps/sig-cli/4944-kustomize-wasm-plugin/README.md

```diff
@@ -0,0 +1,878 @@
+  - WASI is support to stdin/stdout, so We reuse the current interface
+    definitions and users can migrate with minimal changes.
+- Now,
+  [Wasm binary can be distributed with OCI registry](https://tag-runtime.cncf.io/wgs/wasm/deliverables/wasm-oci-artifact/).
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+1. WASI Specific is not yet GA. Any disruptive changes in the future will incur
+   user transition costs.
+1. The Wasm OCI Artifact specification is a relatively new standard in the OCI
+   ecosystem. Maybe not mature enough.
+1. Containaized KRM function is alpha feature but that used some user.
+   - Provide container to exec conversion tools.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+Implement with [wazero](https://github.com/tetratelabs/wazero) to WASM runtime
+like the
+[kube-scheduler-wasm-extension](https://github.com/kubernetes-sigs/kube-scheduler-wasm-extension)
+project.
+
+### Interface
+
+Using the WASI to plugin interface.
+
+##### with Wasm binary file
+
+```yaml
+apiVersion: transformers.example.co/v1
+kind: App
+metadata:
+  name: example-app
+  annotations:
+    config.kubernetes.io/function: |
+      wasip1:
+        file: ../bin/fnapp.wasm
+spec:
+  port: 8080
+```
+
+##### with OCI registry
+
+```yaml
+apiVersion: transformers.example.co/v1
+kind: App
+metadata:
+  name: example-app
+  annotations:
+    config.kubernetes.io/function: |
+      wasip1:
+        image: ghcr.io/koba1t/krm-fn-app:v0.0.3-wasm
+spec:
+  port: 8080
+```
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[ ] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes
+necessary to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
```

Comment:
For kustomize, I'm certain we can just skip interrogation and e2e tests, but listing units would be reasonable. 
Something like:
```
- sigs.k8s.io/kustomize/kustomize/v5/commands/build	- 2025-06-01 - 79.4
- sigs.k8s.io/kustomize/kustomize/v5/commands/create - 2025-06-26 - 84.9
etc...
```
Just make sure to list only the directories you'll be modifying as part of this functionality. There's no need to list all of them :wink: